CardMatchingGameWithBean/Model/CardMatchingGame.m

```diff
@@ -45,48 +49,92 @@ -(instancetype) initWithCardCount:(NSUInteger)count usingDeck:(Deck *)deck
     return self;
 }
 
-
--(Card *)cardAtIndex:(NSUInteger)index
+- (Card *)cardAtIndex:(NSUInteger)index
 {
     return (index < [self.cards count]) ? self.cards[index] : nil;
 }
 
+// Assignment 2, Task 3
+// modified chooseCardAtIndex: method to include the segmentedControl index that
+// indicates which game the user want to play
 
-static const int MISMATCH_PENALTY = 2;
-static const int MATCH_BONUS = 4;
-static const int COST_TO_CHOOSE = 1;
-
-- (void)chooseCardAtIndex:(NSUInteger)index
+- (void)chooseCardAtIndex:(NSUInteger)cardIndex
+ getSegmentedControlIndex:(NSInteger)segmentedControlIndex
 {
-    Card *card = [self cardAtIndex:index];
-    NSLog(@"%@ we have a card this is going to match", card);
+    // Assignment 2, Task 3
+    // modify cardAtIndex: below to take cardIndex
+    // I just renamed index to cardIndex, so we don't
+    // get confused with segmentedControl's index and the
+    // card's index
+
+    // other edits: since we are only going to be worried about
+    // 2 or 3 card game, I modified this function to differentiate between
+    // each game and process the info (segmentIndex, cardIndex) with respect
+    // to the specific match game.
+
+    Card *card = [self cardAtIndex:cardIndex];
+    [self.chosenCards addObject:card];
+    [self scoreGame:@"picked a card"];
     
-    if (!card.isMatched) {
-        if (card.isChosen) {
+    if (!card.matched) {
+        if (card.chosen) {
             card.chosen = NO;
+            [self.chosenCards removeObject:card];
         } else {
-            for (Card *otherCard in self.cards) {
-                if (otherCard.isChosen && !otherCard.isMatched) {
-                    int matchScore = [card match:@[otherCard]];
-                    if (matchScore) {
-                        self.score += (matchScore * MATCH_BONUS);
-                        card.matched = YES;
-                        otherCard.matched = YES;
-                    } else {
-                        self.score -= MISMATCH_PENALTY;
-                        otherCard.chosen = NO;
-                    }
-                    break;
+            card.chosen = YES;
+            NSInteger cardsInArray =
+                [self _differentiateTwoAndThreeCardGames:
+                          segmentedControlIndex];
+            if (cardsInArray) {
+                self.matchScore = [card match:self.chosenCards];
+                if (self.matchScore) {
+                    [self matchCards:self.chosenCards];
+                } else if (self.matchScore == 0) {
+                    [self unmatchCards:self.chosenCards];
```

Comment:
`0` will automatically evaluate to `NO` in an if statement, so you don't have to use else if == 0, you can just use else